inter/block.go

```diff
@@ -15,12 +15,25 @@ type Block struct {
 	SkippedTxs  []uint32      // indexes of skipped txs, starting from first tx of first event, ending with last tx of last event
 	GasUsed     uint64
 	Root        hash.Hash
+	prevRandao  common.Hash
 }
 
 func (b *Block) EstimateSize() int {
 	return (len(b.Events)+len(b.InternalTxs)+len(b.Txs)+1+1)*32 + len(b.SkippedTxs)*4 + 8 + 8
 }
 
+func (b *Block) SetPrevRandao(prevrandao common.Hash) {
+	b.prevRandao = prevrandao
+}
+
+func (b *Block) GetPrevRandao() common.Hash {
+	// Dummy blocks never run through block processing
+	if b.prevRandao == (common.Hash{}) {
+		b.prevRandao = ComputePrevRandao(b.Events)
+	}
+	return b.prevRandao
```

Comment:
I would remove the `prevRandao` cache value and just always compute it based on `b.Events` when needed until I see this to be an actual cause for performance issues. The reason: the `Block` type is not protected against arbitrary modifications. Somebody may change the `Events` list without resetting `prevRandao` and it would be very difficult to figure out that these two fields are no longer aligned.

Reply:
I think that computing `prevrandao` as part of struct creation would work as well. The events are set when creating the struct, and I think they are not changed during the lifecycle of `block` instance.

Reply:
Caching removed.

Reply:
As @HerbertJordan says, since the events are public, it can be changed hence the `Prevrandao` value would not correspond with it.